File: dashlane-project/Dashlane/src/main/java/com/dashlane/async/broadcasts/InstallReceiver.java
package com.dashlane.async.broadcasts;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.adjust.sdk.AdjustReferrerReceiver;
import com.dashlane.analytics.install.InstallTrackingManager;
import com.dashlane.async.BroadcastManager;
import com.dashlane.dagger.singleton.SingletonProvider;
import com.dashlane.debug.DeveloperUtilities;
import com.dashlane.preference.GlobalPreferencesManager;
import com.dashlane.util.Constants;

import java.util.Set;

public class InstallReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context receiverContext, final Intent intent) {
        
        SingletonProvider.getThreadHelper().runOnBackgroundThread(() -> {

            
            final Context context = SingletonProvider.getContext();
            GlobalPreferencesManager preferencesManager = SingletonProvider.getGlobalPreferencesManager();
            InstallTrackingManager installTracker = new InstallTrackingManager();
            installTracker.installEvent(context, intent);
            
            if (DeveloperUtilities.systemIsInDebug(context)) {
                return;
            }

            BroadcastManager.removeAllBufferedIntent();

            if (!DeveloperUtilities.systemIsInDebug(context)) {
                sendAdjustEvent(context, intent);
            }
            if (intent == null) {
                return;
            }
            if (intent.getExtras() != null) {
                Set<String> keys = intent.getExtras().keySet();
                StringBuilder result = new StringBuilder();
                for (String string : keys) {
                    result.append(string)
                          .append("=")
                          .append(intent.getStringExtra(string))
                          .append("&");
                }
            }

            preferencesManager.putBoolean(Constants.MARKETING.SHOULD_SEND_REPORTS, true);

            if (intent.hasExtra("referrer")) {
                preferencesManager.putString(Constants.MARKETING.REFFERAL_STRING,
                                             intent.getExtras().getString("referrer"));
            }
        });

    }

    private void sendAdjustEvent(Context context, Intent intent) {
        SingletonProvider.getAdjustWrapper().initIfNeeded(context);
        new AdjustReferrerReceiver().onReceive(context, intent);
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/async/broadcasts/InstallReceiver.kt
package com.dashlane.async.broadcasts

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.adjust.sdk.AdjustReferrerReceiver
import com.dashlane.analytics.install.InstallTrackingManager
import com.dashlane.async.SyncBroadcastManager
import com.dashlane.debug.DeveloperUtilities.systemIsInDebug
import com.dashlane.logger.AdjustWrapper
import com.dashlane.preference.GlobalPreferencesManager
import com.dashlane.util.inject.qualifiers.ApplicationCoroutineScope
import com.dashlane.util.inject.qualifiers.DefaultCoroutineDispatcher
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import javax.inject.Inject

@AndroidEntryPoint
class InstallReceiver : BroadcastReceiver() {
    @Inject
    lateinit var globalPreferencesManager: GlobalPreferencesManager

    @Inject
    lateinit var adjustWrapper: AdjustWrapper

    @Inject
    lateinit var syncBroadcastManager: SyncBroadcastManager

    @Inject
    @ApplicationCoroutineScope
    lateinit var applicationCoroutineScope: CoroutineScope

    @Inject
    @DefaultCoroutineDispatcher
    lateinit var defaultCoroutineDispatcher: CoroutineDispatcher

    @Inject
    lateinit var installTrackingManager: InstallTrackingManager

    override fun onReceive(context: Context, intent: Intent) {
        applicationCoroutineScope.launch {
            withContext(defaultCoroutineDispatcher) {
                
                installTrackingManager.installEvent(intent)

                
                if (systemIsInDebug(context)) {
                    return@withContext
                }
                syncBroadcastManager.removePasswordBroadcastIntent()
                if (!systemIsInDebug(context)) {
                    sendAdjustEvent(context, intent)
                }
            }
        }
    }

    private fun sendAdjustEvent(context: Context, intent: Intent) {
        adjustWrapper.initIfNeeded(context)
        AdjustReferrerReceiver().onReceive(context, intent)
    }
}
